File: app/src/main/java/me/tagavari/airmessage/messaging/StickerInfo.java
package me.tagavari.airmessage.messaging;

import android.os.Parcel;
import android.os.Parcelable;

import java.io.File;

public class StickerInfo implements Parcelable {
	//Creating the sticker values
	private final long localID;
	private final String guid;
	private final String sender;
	private final long date;
	private final File file;
	
	public StickerInfo(long localID, String guid, String sender, long date, File file) {
		this.localID = localID;
		this.guid = guid;
		this.sender = sender;
		this.date = date;
		this.file = file;
	}
	
	public long getLocalID() {
		return localID;
	}
	
	public String getGuid() {
		return guid;
	}
	
	public String getSender() {
		return sender;
	}
	
	public long getDate() {
		return date;
	}
	
	public File getFile() {
		return file;
	}
	
	@Override
	public int describeContents() {
		return 0;
	}
	
	@Override
	public void writeToParcel(Parcel out, int flags) {
		out.writeLong(localID);
		out.writeString(guid);
		out.writeString(sender);
		out.writeLong(date);
		out.writeString(file.getPath());
	}
	
	public static final Parcelable.Creator<StickerInfo> CREATOR = new Parcelable.Creator<StickerInfo>() {
		public StickerInfo createFromParcel(Parcel in) {
			return new StickerInfo(in);
		}
		
		public StickerInfo[] newArray(int size) {
			return new StickerInfo[size];
		}
	};
	
	private StickerInfo(Parcel in) {
		localID = in.readLong();
		guid = in.readString();
		sender = in.readString();
		date = in.readLong();
		file = new File(in.readString());
	}
}

File: common/me/tagavari/airmessage/common/messaging/StickerInfo.kt
package me.tagavari.airmessage.common.messaging

import android.os.Parcel
import android.os.Parcelable
import androidx.compose.runtime.Immutable
import java.io.File

@Immutable
class StickerInfo : Parcelable {
	//Creating the sticker values
	val localID: Long
	val guid: String?
	val sender: String?
	val date: Long
	val file: File
	
	constructor(localID: Long, guid: String?, sender: String?, date: Long, file: File) {
		this.localID = localID
		this.guid = guid
		this.sender = sender
		this.date = date
		this.file = file
	}
	
	override fun describeContents(): Int {
		return 0
	}
	
	override fun writeToParcel(out: Parcel, flags: Int) {
		out.writeLong(localID)
		out.writeString(guid)
		out.writeString(sender)
		out.writeLong(date)
		out.writeString(file.path)
	}
	
	private constructor(parcel: Parcel) {
		localID = parcel.readLong()
		guid = parcel.readString()
		sender = parcel.readString()
		date = parcel.readLong()
		file = File(parcel.readString()!!)
	}
	
	companion object {
		@JvmField val CREATOR: Parcelable.Creator<StickerInfo> = object : Parcelable.Creator<StickerInfo> {
			override fun createFromParcel(parcel: Parcel) = StickerInfo(parcel)
			
			override fun newArray(size: Int) = arrayOfNulls<StickerInfo>(size)
		}
	}
}